### PR TITLE
make LOGDEBUG default for addons

### PR DESCRIPTION
**Pull request by ronie**
addons should only use the LOGDEBUG level for logging, so let's make it the default.

@jimfcarroll for review

### CODE REVIEW

File: tools/codegenerator/example/native/ModuleXbmc.h [hunk 1]
```diff
@@ -1 +1 @@
-void log(const char* msg, int level = LOGNOTICE);
+void log(const char* msg, int level = LOGDEBUG);
```
Comment (tamland) [inline tools/codegenerator/example/native/ModuleXbmc.h]: This is an example project for the codegenerator tool. Nothing to do with the other `log`

**Comment by tamland** [comment]
Otherwise it should be good

File: tools/codegenerator/example/native/ModuleXbmc.h (hunk 1 above)
Comment (ronie) [inline tools/codegenerator/example/native/ModuleXbmc.h]: thx, i'll drop that change then.

**Comment by jimfcarroll** [comment]
Sorry this is late. I missed the first note. It looks like it will do what you want it to do. Do you really want to change the default log level?

If you really want addons to ONLY be able to log at a debug level then why not remove the option to log at a different level?

**Comment by ronie** [comment]
yes, we've been asking addon devs for a few years to only use the debug loglevel.
so it makes sense to use it as a default.

i don't think it's a good idea the remove the loglevel specification all together.
some addons catch exceptions and log them to the logfile,
it's perfectly fine to use LOGERROR in those occasions.

**Comment by KenV99** [comment]
Please consider creating an option under component specific logging for addons.
Rationale:
For intermittent bugs, the user would need to turn on full debugging with the OSD which interferes with watching videos, etc.
OR:
A less desirable alternative would be to at least create a way within the GUI to turn the debug OSD off while still keeping the debug level at 1 without requiring editing the advancedsettings.xml file.
Thanks for your consideration.

**Pull request merged**